### PR TITLE
Add paddle operator proposal to kubeflow community.

### PR DESCRIPTION
**Pull request by tizhou86**
Add paddle operator proposal to kubeflow community.

### CODE REVIEW

File: proposals/paddle-operator-proposal.md
Comment (terrytangyuan) [inline proposals/paddle-operator-proposal.md:230]: This makes sense. What about the implementation of the operator? Could you leverage https://github.com/kubeflow/common's interface that's used for other operators?

Comment (terrytangyuan) [inline proposals/paddle-operator-proposal.md:230]: This makes sense. What about the implementation of the operator? Could you leverage https://github.com/kubeflow/common's interface that's used for other operators?

Comment (tizhou86) [reply proposals/paddle-operator-proposal.md:230]: We are using kube-builder as the skeleton for the paddle-operator, part of the kubeflow common components are not necessary for our project, like job controller etc. We'll see whether we can leverage on kubeflow common for advanced features.

**Comment by terrytangyuan** [comment]
cc @kubeflow/wg-training-leads

**Review by gaocegege — APPROVED** [review]

File: proposals/paddle-operator-proposal.md
Comment (kuizhiqing) [reply proposals/paddle-operator-proposal.md:230]: Thanks for your reply, we are aware of your concern, technically, we are building our operator with the newest version of kubebuilder which take care of almost all the staff like Informer/Indexer/clienset etc. It leaves out the Reconcile function to be implemented, it also provides CRUD operations directly on resources with context, which means that the kubeflow/common may not necessary in this circumstances.

**Review by terrytangyuan — APPROVED** [review]
This LGTM from me. Please take another look @kubeflow/wg-training-leads @kubeflow/wg-automl-leads as well.

@tizhou86 @kuizhiqing Are you planning to transfer the existing repo https://github.com/PaddleFlow/paddle-operator to Kubeflow org once this is approved? @Bobgy @theadactyl Any questions or concerns on this?

File: proposals/paddle-operator-proposal.md
Comment (PatrickXYS) [inline proposals/paddle-operator-proposal.md:225]: All link expired

**Review by PatrickXYS — COMMENTED** [review]
Awesome! 

Only nits comment

**Comment by Jeffwan** [comment]
We briefly talked about this story in AutoML and training bi-weekly meeting. People are interested in support plan as well. For example, if there's active maintainers for this project. If it want to be part of kubeflow release, etc.

File: proposals/paddle-operator-proposal.md
Comment (tizhou86) [reply proposals/paddle-operator-proposal.md:225]: Since we've just merged the new refactoring, I'll fix it soon.

**Comment by tizhou86** [comment]
> This LGTM from me. Please take another look @kubeflow/wg-training-leads @kubeflow/wg-automl-leads as well.
> 
> @tizhou86 @kuizhiqing Are you planning to transfer the existing repo https://github.com/PaddleFlow/paddle-operator to Kubeflow org once this is approved? @Bobgy @theadactyl Any questions or concerns on this?

Yes, Paddle-operator is one of our ecosystem projects that we want to transfer to kubeflow org for maintenance, It's approved by the Paddle team already, we think the open governance will be good for the Paddle ecosystem projects.

**Comment by tizhou86** [comment]
> We briefly talked about this story in AutoML and training bi-weekly meeting. People are interested in support plan as well. For example, if there's active maintainers for this project. If it want to be part of kubeflow release, etc.

Great! If possible, we can do a presentations on kubeflow weekly meeting, and then we can discuss about the future plan in the meeting.

**Comment by Jeffwan** [comment]
@gaocegege  Last WG-training meeting was CN time friendly and I miss it. Any conclusion?

**Comment by gaocegege** [comment]
We think we can accept it. If we decide to merge into one controller, we can refactor the paddle operator later.

WDYT

**Comment by Jeffwan** [comment]
> We think we can accept it. If we decide to merge into one controller, we can refactor the paddle operator later.
> 
> WDYT

Agree.  Let's kick off the transfer process. 

/lgtm
/approve

@Bobgy  @theadactyl  @gaocegege @terrytangyuan and I help review the proposal from WG-Training side and it looks good to us. Can you help take a review?

**Comment by gaocegege** [comment]
/cc @Bobgy 

Could you please help us create the repository for the paddle-operator?

**Comment by tizhou86** [comment]
Hi Bob, please let us know if anything should be done before the transfer process, thanks!

**Comment by tizhou86** [comment]
> Hi Bob, please let us know if anything should be done before the transfer process, thanks!

The original repository is: https://github.com/PaddleFlow/paddle-operator , the new repository I suppose will be : https://github.com/kubeflow/paddle-operator , we've recently added more user guides and documentations for developers.

**Comment by Bobgy** [comment]
Sorry for the delay, @kubeflow/wg-training-leads.

I discussed with @kubeflow/project-steering-group and we would suggest merging all training operators into one monorepo. How do feel about that?

The rationale:
we need to do a complex release process each time a training operator repo is created. It seems that the number of frameworks will only keep increase in the future. However, if I understand correctly, most operators share quite some stuff in common. So I think it might be a lower effort thing for both of us to create one central training operators repo once, and keep adding framework-specific operators to it.

**Comment by Bobgy** [comment]
I think the proposal also relates to https://github.com/kubeflow/community/pull/512

**Comment by terrytangyuan** [comment]
@Bobgy Please see our discussion on merging the operators in https://github.com/kubeflow/common/issues/103.

**Comment by Bobgy** [comment]
@terrytangyuan regardless of whether operators are merged, what about using a monorepo?
I'm not talking about the engineering process for release, but actually Google's internal release process, any new repo in Kubeflow org needs to go through Google release process and we want to avoid this by using a monorepo (anyway, only in case when monorepo makes sense for Training WG).

Looks like we can wait for https://github.com/kubeflow/common/issues/103#issuecomment-831044131 to see more updates?

**Comment by Jeffwan** [comment]
> any new repo in Kubeflow org needs to go through Google release process

em. Since this is internal process and we don't have any clues, could you help us understand the efforts? It helps WG leads to make deliberate decision at the time we have these kind of repo requests.

**Comment by tizhou86** [comment]
Yeah, we are really looking forward to have paddle operator to integrated into Kubeflow community, some of our end users are building the PaddlePaddle deep learning system leveraging on Kubeflow and paddle operator, we want to make it more straightforward to our users. And with the impact of the PaddlePaddle community(15.4k for the [paddle main repo](https://github.com/PaddlePaddle/Paddle) and 50k+for the [paddle org projects](https://github.com/PaddlePaddle)), we think it will be a win-win solution for both community.

**Comment by Bobgy** [comment]
Sorry for the long delay, let me give an update:
1. I have learned what the process for adopting a new repo roughly looks like, but I'll have to do it once to understand how much efforts we need each time. Due to policy issues, I cannot share more details.
2. I had some discussions within PSG, but haven't reached a final conclusion yet. I'll update again next week.

**Comment by Bobgy** [comment]
Hi all, @kubeflow/project-steering-group agreed to accept this proposal and start a new repo for paddle-operator without waiting for potential merge of Training WG operators.

@kubeflow/wg-training-leads may I confirm that you are still willing to accept the proposal too, right? because it's been a while.

If everyone agrees, I can start working with @tizhou86 on the process to adopt the paddle-operator repo.

**Comment by terrytangyuan** [comment]
+1 from me. @kubeflow/wg-training-leads Any concerns?

**Comment by johnugeorge** [comment]
+1

**Comment by gaocegege** [comment]
SGTM

**Comment by tizhou86** [comment]
Great, thanks @Bobgy ! Please let me know if anything is needed from PaddlePaddle team.

**Comment by Bobgy** [comment]
/lgtm
/approve
Let's get the proposal in first.

**Pull request merged**
